src/index.ts

```diff
@@ -1,50 +1,22 @@
-/**
- * Return a function that will copy properties from
- * one object to another excluding any originally
- * listed. Returned function will create a new `{}`.
- *
- * @param {String} excluded properties ...
- * @return {Function}
- */
-function exclude(...args: string[]): (a: {}, b?: {}) => Record<string, unknown> {
-  var excludes = args
-
-  function excludeProps (res: Record<string, unknown>, obj: Record<string, unknown>) {
-    Object.keys(obj).forEach(function (key) {
-      if (!~excludes.indexOf(key)) res[key] = obj[key];
-    });
-  }
-
-  return function extendExclude () {
-    var args = [].slice.call(arguments)
-      , i = 0
-      , res = {};
-
-    for (; i < args.length; i++) {
-      excludeProps(res, args[i]);
-    }
-
-    return res;
-  };
-};
-
 export default class AssertionError<T> extends Error {
   name = 'AssertionError'
   showDiff: boolean
   [key: string]: any
 
-  constructor(message: string, _props?: T, ssf?: Function) {
-    super()
-    var extend = exclude('name', 'message', 'stack', 'constructor', 'toJSON')
-      , props = extend(_props || {});
+  constructor(message: string, props?: T, ssf?: Function) {
+    super(message)
 
     // default values
     this.message = message || 'Unspecified AssertionError';
     this.showDiff = false;
 
     // copy from properties
     for (var key in props) {
-      this[key] = props[key];
+      const propertiesNotToSet = ['name', 'message', 'stack', 'constructor', 'toJSON']
```

Comment:
This could be outside of the class scope

Reply:
What do you think about 8090b90? I was thinking that this mechanism was originally to make sure we don't overwrite the properties of the error. With that in mind we can just check if the property is defined and not overwrite it if it is defined.